classes/validation/ValidatorFactory.inc.php

```diff
@@ -308,8 +308,8 @@ public static function required($validator, $action, $requiredProps, $multilingu
                         }
                     }
                 } else {
-                    if (($action === EntityWriteInterface::VALIDATE_ACTION_ADD && self::isEmpty($props[$requiredProp])) ||
-                            ($action === EntityWriteInterface::VALIDATE_ACTION_EDIT && array_key_exists($requiredProp, $props) && self::isEmpty($props[$requiredProp]))) {
```

Comment:
I'm not sure how the new code captures the distinction between `ADD` and `EDIT` just by checking for the presence of `$object`...

Reply:
My thinking here is that an object can only exist when validating an existing object. If validating properties for a new object, the object doesn't exist yet. The method is called by passing `null` in for the `$object`:

https://github.com/NateWr/pkp-lib/blob/i5328_service_providers/api/v1/submissions/PKPSubmissionHandler.inc.php#L382

This is maybe confusing because we never validate whole objects. We only validate the properties that are currently being modified. I found this useful because -- especially for the context -- there are a lot of different places that an object can be edited. So if validation of an object fails, but it fails based on some pre-existing data (not the data that I'm editing), it's impossible to provide a good UX. Here's an example of where this might be useful:

> A context has a `urlPath` of `Capitals`. The service provider wants to enforce consistent URLs, so enables a plugin that modifies the `urlPath` validation to only support lower-case letters. The journal editor tries to edit the journal's about text, but a mysterious error message doesn't let them edit the text.

So the validation only runs against the properties being modified, but it's still really useful to have the object being modified if it exists. We ran into this problem with the language settings validation that @henriqueramos worked on. To know what were valid `supportedFormLocales` we needed to know what the context's `supportedLocales` setting was.

So maybe passing the `$object` isn't a very intuitive way to do this. But I thought it would maybe streamline the process a bit and reduce reliance on the user knowing a specific constant to use.

Reply:
This makes sense (and we're using object presence/absence in similar ways pretty frequently) and seems a more intuitive idiom than introducing NAMED_CONSTANTS though the latter may still be necessary in places where we need more granularity (e.g. the file capabilities). :+1: